### PR TITLE
Refactor for 3.0.0

### PR DESCRIPTION
**Pull request by cheeseplus**
Signed-off-by: Seth Thomas <sthomas@chef.io>

### Description

This PR broadly fixes the cookbook into a state that is working and passes tests.

- removal of erroneous and derived attributes
- removal of unnecessary recipes
- rename `server` recipe to `default`
- rename attributes for clarity, i.e. `ssl_cert` -> `tls_cert`
- remove `cn=config` generation as it never actually worked (init scripts still default to conf file)
- `provider` and `consumer` terminology adopted
- ensure package doesn't autostart as part of dpkg install on Debian platforms
- OpenSUSE and Fedora support
- init/default/sysconfig files for platforms as necessary
- LDAPS support is now toggleable due to above

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [X] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -77,45 +77,14 @@ platforms:
...
-      basedn: "dc=example, dc=com"
-      server: "ldap.example.com"
-      preseed_dir: "/tmp/does_not_exist"
-      ssl_cert_source_cookbook: "openldap-test"
-      ssl_cert_source_path: "ldap.example.com.pem"
-      ssl_key_source_cookbook: "openldap-test"
-      ssl_key_source_path: "ldap.example.com.key"
-- name: unmanaged_ssl
+  - recipe[openldap-test::default]
+  - recipe[openldap::default]
```
Comment (tas50) [inline .kitchen.dokken.yml]: The test recipes should just do the include on this. That way they look more like true wrapper cookbooks and we move a bit more logic out of TK configs

File: .kitchen.dokken.yml [hunk 2]
```diff
@@ -77,45 +77,14 @@ platforms:
...
+- name: tls_enabled
   run_list:
     - recipe[openldap-test::ssl]
-    - recipe[openldap::server]
+    - recipe[openldap::default]
   attributes:
     openldap:
-      rootpw: "{SSHA}6BjlvtSbVCL88li8IorkqMSofkLio58/" # secretsauce
```
Comment (tas50) [inline .kitchen.dokken.yml]: This red is beautiful

File: .travis.yml [hunk 3]
```diff
@@ -19,10 +19,10 @@ services: docker
...
 env:
   matrix:
-  - INSTANCE=server-centos-6
-  - INSTANCE=server-centos-7
-  - INSTANCE=server-ubuntu-1404
-  - INSTANCE=server-ubuntu-1604
+  - INSTANCE=default-centos-6
+  - INSTANCE=default-centos-7
+  - INSTANCE=default-ubuntu-1404
+  - INSTANCE=default-ubuntu-1604
```
Comment (tas50) [inline .travis.yml]: if Fedora / openSUSE work we should get them in here. It's not a blocker on 3.0 though

File: metadata.rb [hunk 4]
```diff
@@ -2,19 +2,18 @@
...
+version           '3.0.0'
 
-recipe            'openldap',         'Empty, use one of the other recipes'
-recipe            'openldap::server', 'Set up openldap to be a standalone server'
-recipe            'openldap::master', 'Sets up openldap as a replication master'
-recipe            'openldap::slave',  'Sets up openldap to replicate to a specified master'
+recipe            'openldap::default', 'Install and configure OpenLDAP'
 
-%w(ubuntu debian freebsd redhat centos amazon scientific oracle).each do |os|
+%w(ubuntu debian freebsd redhat centos opensuse).each do |os|
```
Comment (tas50) [inline metadata.rb]: scientific and oracle are kinda a given. Amazon I'm sure has some funky package that breaks the world

File: .kitchen.dokken.yml (hunk 1 above)
Comment (cheeseplus) [inline .kitchen.dokken.yml]: I'd had that thought too so I'm down to change that

File: metadata.rb (hunk 4 above)
Comment (cheeseplus) [inline metadata.rb]: Happy to add those back, was mostly worried about amazon

File: .travis.yml (hunk 3 above)
Comment (cheeseplus) [inline .travis.yml]: We also have debian too which is missing from the matrix

**Pull request merged**